### PR TITLE
Requesting a pull to mozilla:master from k88hudson:t1787

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/bb3f2a949dca33c/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/bb3f2a949dca33c/output.txt

Total script time: 0.90 mins

File: src/ui/ui.js [hunk 1]
```diff
@@ -60,15 +60,20 @@ define( [ "core/eventmanager", "./toggler",
...
 
-      for ( icon in icons ) {
-        if ( icons.hasOwnProperty( icon ) ) {
-          img = new Image();
-          img.id = icon + "-icon";
-          img.src = resourcesDir + icons[ icon ];
+      plugins.forEach( function( plugin ) {
+          path = plugin.icon;
 
+          if( !path ) {
```
Comment (mjschranz) [inline src/ui/ui.js]: space after if

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/c29db857cc84fda/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/c29db857cc84fda/output.txt

Total script time: 0.55 mins

Server is running on http://pr797.butyr.org/templates/basic/
Run the Test Suite: http://pr797.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/b780a320c55f369/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/b780a320c55f369/output.txt

Total script time: 1.10 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/19eda922237d990/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/19eda922237d990/output.txt

Total script time: 0.79 mins

Server is running on http://pr797.butyr.org/templates/basic/
Run the Test Suite: http://pr797.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/82ed15c4b5e70b8/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/82ed15c4b5e70b8/output.txt

Total script time: 2.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/85ad6ced47228b8/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/85ad6ced47228b8/output.txt

Total script time: 0.90 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/150c17f919c6d57/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/150c17f919c6d57/output.txt

Total script time: 0.86 mins

Server is running on http://pr797.butyr.org/templates/basic/
Run the Test Suite: http://pr797.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/a478c597b1661b7/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/a478c597b1661b7/output.txt

Total script time: 1.18 mins

**Pull request merged**